src/adapter/templates/getStringyProps.ts

```diff
@@ -2,31 +2,66 @@
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
-import { remoteFunction } from '.';
+import { templateFunction } from '.';
 
 /**
  * Gets a mapping of property names with a custom `.toString()` method
  * to their string representations.
  */
-export const getStringyProps = remoteFunction(function (this: unknown, maxLength: number) {
+export const getStringyProps = templateFunction(function (
+  this: unknown,
+  maxLength: number,
+  customToString: (defaultRepr: string) => unknown,
+) {
   const out: Record<string, string> = {};
+  const defaultPlaceholder = '<<default preview>>';
   if (typeof this !== 'object' || !this) {
     return out;
   }
 
   for (const [key, value] of Object.entries(this)) {
+    if (customToString) {
+      try {
+        const repr = customToString.call(value, defaultPlaceholder);
+        if (repr !== defaultPlaceholder) {
+          out[key] = String(repr);
+          continue;
+        }
+      } catch (e) {
+        out[key] = `${e} (couldn't describe ${key})`;
```

Comment:
Does this need localization?

Reply:
Eh, probably. This is awkward since this code is executed in the runtime.